conv/abci/block.go

```diff
@@ -121,28 +121,34 @@ func ToABCIBlockMeta(block *types.Block) (*tmtypes.BlockMeta, error) {
 // This function only converts fields that are available in Dymint commit.
 // Other fields (especially ValidatorAddress and Timestamp of Signature) has to be filled by caller.
 func ToABCICommit(commit *types.Commit, header *types.Header) *tmtypes.Commit {
+	headerHash := header.Hash()
 	tmCommit := tmtypes.Commit{
 		Height: int64(commit.Height),
 		Round:  0,
 		BlockID: tmtypes.BlockID{
-			Hash: commit.HeaderHash[:],
+			Hash: headerHash[:],
```

Comment:
why u need the `commit *types.Commit` argument? seems u take everything from the header itself

Reply:
this line for example `if len(commit.TMSignature.Signature) == 0 {` is using the `commit` field and can't use the header field.